libexec/bootstrap-scripts/deffile-sections.sh

```diff
@@ -373,7 +373,9 @@ for app in ${SINGULARITY_ROOTFS}/scif/apps/*; do
     if [ -d "$app" ]; then
 
         app="${app##*/}"
-        appvar=(`echo $app | sed -e "s/-/_/g"`)
+
+        # Replace all - or . with an underscore
+        appvar=(`echo $app | sed -e "s/-/_/g" | sed -e "s/[.]/_/g"`)
```

Comment:
Not that it makes a huge difference, but `-e` can be used to specify multiple commands to a single sed invocation [1]. Another alternative is changing the pattern to match both characters.

```
sed -e "s/-/_/g" -e "s/[.]/_/g"
sed "s/[-.]/_/g"
```

[1] http://www.grymoire.com/Unix/Sed.html#uh-13